common/src/voxel_math.rs

```diff
@@ -56,9 +58,25 @@ pub enum CoordSign {
 impl CoordSign {
     /// Iterates through the two possible coordinate directions
     pub fn iter() -> impl ExactSizeIterator<Item = Self> {
-        [CoordSign::Plus, CoordSign::Minus]
-            .iter()
-            .copied()
+        [CoordSign::Plus, CoordSign::Minus].iter().copied()
+    }
+}
+
+impl std::ops::Mul for CoordSign {
+    type Output = CoordSign;
+
+    fn mul(self, rhs: Self) -> Self::Output {
+        if self == rhs {
+            CoordSign::Plus
+        } else {
+            CoordSign::Minus
+        }
```

Comment:
style nit: this would be more concise as a `match self == rhs`

Reply:
That does look nicer. I wonder if it's worth defaulting to that in situations where I would use the ternary operator in other programming languages. I think there might be some circumstances where that's less readable, so I'll at least keep in mind that it's an option.